cromwell_tools/cromwell_tools.py

```diff
@@ -1,34 +1,144 @@
+                secrets = json.load(f)
+                cromwell_username = secrets['cromwell_user']
+                cromwell_password = secrets['cromwell_password']
+    return cromwell_username, cromwell_password
+
+
+def get_workflow_statuses(
+        ids, cromwell_url, cromwell_user=None, cromwell_password=None, secrets_file=None):
+    """ given a list of workflow ids, query cromwell url for their statuses
+
+    :param list ids:
+    :param str cromwell_url:
+    :param str cromwell_user:
+    :param str cromwell_password:
+    :param str secrets_file:
+    :return list: list of workflow statuses
+    """
+    cromwell_user, cromwell_password = harmonize_credentials(
+        secrets_file, cromwell_user, cromwell_password)
+    statuses = []
+    for id_ in ids:
+        full_url = cromwell_url + '/api/workflows/v1/{0}/status'.format(id_)
+        auth = requests.auth.HTTPBasicAuth(cromwell_user, cromwell_password)
+        response = requests.get(full_url, auth=auth)
+        if response.status_code != 200:
+            print('Could not get status for {0}. Cromwell at {1} returned status {2}'.format(
+                id_, cromwell_url, response.status_code))
+            statuses.append('Unknown')
+        else:
+            response_json = response.json()
+            status = response_json['status']
+            statuses.append(status)
+            timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+            print('{0} workflow {1}: {2}'.format(timestamp, id_, status))
+    return statuses
+
+
+def wait_until_workflow_completes(
+        cromwell_url, workflow_ids, timeout_minutes, poll_interval_seconds=30, cromwell_user=None,
+        cromwell_password=None, secrets_file=None):
+    """
+    given a list of workflow ids, wait until cromwell returns successfully for each status, or
+    one of the workflows fails or is aborted.
+
+    :param list workflow_ids:
+    :param int timeout_minutes:
+    :param int poll_interval_seconds: number of seconds between checks for workflow completion
+      (default 30)
+    :param str cromwell_url:
+    :param str cromwell_user:
+    :param str cromwell_password:
+    :param str secrets_file:
+    :return:
+    """
+    cromwell_user, cromwell_password = harmonize_credentials(
+        secrets_file, cromwell_user, cromwell_password)
+    start = datetime.now()
+    timeout = timedelta(minutes=int(timeout_minutes))
+    while True:
+        if datetime.now() - start > timeout:
+            msg = 'Unfinished workflows after {0} minutes.'
+            raise Exception(msg.format(timeout))
+        statuses = get_workflow_statuses(workflow_ids, cromwell_url, cromwell_user, cromwell_password)
+        all_succeeded = True
+        for i, status in enumerate(statuses):
+            if status in _failed_statuses:
+                raise Exception('Stopping because workflow {0} {1}'.format(workflow_ids[i], status))
+            elif status != 'Succeeded':
+                all_succeeded = False
```

Comment:
I'd add a `break` here to avoid unnecessary checks since you want to wait until workflows complete.

Reply:
So, as I copied it over, I think this is designed to fail on the first failing workflow (see the Exception above). However I agree that the _name_ is misleading.

Reply:
Yea, reading the logic I think things make sense as programmed.